packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -448,15 +488,25 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
         _isChildExpanded(index),
       );
 
-      Widget expandIconContainer = Container(
-        margin: const EdgeInsetsDirectional.only(end: 8.0),
-        child: ExpandIcon(
+      Widget expansionIndicator;
+      if (child.expansionIndicator != null) {
+        expansionIndicator = child.expansionIndicator(
```

Comment:
It might be better to check if it expansionIndicator is a one of the clickable widget, I am not sure if it is even possible. or we can wrap it with gesturedector to make sure it will always respond to tap event.

Another thing to think about is semantics. If you look at the ExpandIcon it disabled onPressed when child.canTapOnHeader is true, it might create some inconsistency if we use expansionIndicator with canTapOnHeader set to true

Reply:
Good point. I think wrapping with a GestureDetector makes the most sense, since I'm not sure we can test for a clickable widget easily, as you said. 

With regards to semantics, this one is a little trickier.. Maybe we could just put in the API docs that they shouldn't  put any GestureDetectors in the builder. If we provide the GestureDetector, then this doesn't seem like it'll be an issue.

Reply:
I ended up using `IgnorePointer` to ignore hit testing if `canTapOnHeader` is set to true